express-delete/index.js

```diff
@@ -0,0 +1,45 @@
+const express = require('express');
+const grades = {
+  12: {
+    id: 12,
+    name: 'Tim Berners-Lee',
+    course: 'HTML',
+    score: 95
+  },
+  47: {
+    id: 47,
+    name: 'Brendan Eich',
+    course: 'JavaScript',
+    score: 100
+  },
+  273: {
+    id: 273,
+    name: 'Forbes Lindsay',
+    course: 'JavaScript',
+    score: 92
+  }
+};
+
+const app = express();
+
+app.get('/api/grades', function (req, res) {
+  const temp = [];
+  for (const x in grades) {
+    temp.push(grades[x]);
+  }
+  res.json(temp);
+});
+
+app.delete('/api/grades/:id', function (req, res) {
+  for (const x in grades) {
+    if (req.params.id === x) {
+      delete grades[x];
+    }
+  }
```

Comment:
You don't need a loop here, you can just directly check if the property exists in the object and delete it if it does.
```JAVASCRIPT
const id = req.params.id;

if (grades[id]) {
  delete grades[id];
}
```